internal/clients/http/azure/create_vm.go

```diff
@@ -33,7 +37,7 @@ const (
 	vmPollFrequency       = 10 * time.Second
 )
 
-func (c *client) BeginCreateVM(ctx context.Context, networkInterface *armnetwork.Interface, vmParams clients.AzureInstanceParams, vmName string) (string, error) {
+func (c *client) BeginCreateVM(ctx context.Context, reservation *models.AzureReservation, networkInterface *armnetwork.Interface, vmParams clients.AzureInstanceParams, vmName string) (string, error) {
```

Comment:
I'd rather not pass the reservation in it's entirety just for this (ideally not at all). It's ofc just a feeling, if you think we should, I'm ok :)

Tho could we add the tags in vmParams? We could add a map there and pass the prepared map of tags :thinking:

WDYT?

Reply:
Good idea, I have missed that.